std/string.d

```diff
@@ -3047,16 +3100,71 @@ if (isConvertibleToString!Range)
     cast(void) stripRight(ws.byUTF!wchar).array;
 }
 
+/// Ditto
+auto stripRight(Range)(Range str, string chars)
+if (isBidirectionalRange!Range && isSomeChar!(ElementEncodingType!Range) &&
+    !isInfinite!Range && !isConvertibleToString!Range)
```

Comment:
bidirectional and infinite not very likely

Reply:
Hold on. How do you strip characters to the *right* of an infinite range?!

E.g., if the infinite range consists of `aaabaaabaaa...` repeated forever, and you call `stripRight` to remove `a`s from the right, how would the code know when to start dropping `a`s? And if `stripRight` was asked to remove *both* `a`s and `b`s, how would the code know if there's a `z` 500 elements into the range so that it shouldn't just return an empty range?

Reply:
The range has `back` and `popBack` and that seems to be all that's needed. It's difficult to define a range that is bidirectional and infinite.

Reply:
Ahh, I misunderstood that you wanted to remove *both* bidirectional and infinite from the constraints.  Removing isInfinite and leaving bidirectional makes sense. :-)

Reply:
P.S. It's not *that* hard to make an infinite range that's also bidirectional; the hard part is coming up with a sensible use case for it. :-P  The traditional infinite range is open to the right, i.e., it has no end. But it's also possible to have a range that's open in the middle, i.e., the front and back parts never meet. Such a range actually *would* work with this code, as is, if we drop `isInfinite`.  Though of course, I've a hard time imagining a real-world use case for this kind of bizarre infinite range!